pkg/virtctl/guestfs/guestfs_test.go

```diff
@@ -149,6 +161,32 @@ var _ = Describe("Guestfs shell", func() {
 			Expect(err).To(HaveOccurred())
 			Expect(err.Error()).Should(Equal(fmt.Sprintf("gid requires the uid to be set")))
 		})
+
+		It("Successfully apply VM's constraints", func() {
+			vmi := libvmi.New(
+				libvmi.WithNamespace(testNamespace),
+				libvmi.WithName("test-vm"),
+				libvmi.WithToleration(v1.Toleration{Key: "tol_key", Value: "tol_val"}),
+				libvmi.WithLabel("label_key", "label_val"),
+				libvmi.WithNodeAffinityForLabel("node_key", "node_val"),
+				libvmi.WithNodeSelector("select_key", "select_val"),
+			)
+			vm := libvmi.NewVirtualMachine(vmi)
+			ctrl := gomock.NewController(GinkgoT())
+			kubecli.GetKubevirtClientFromClientConfig = kubecli.GetMockKubevirtClientFromClientConfig
+			kubecli.MockKubevirtClientInstance = kubecli.NewMockKubevirtClient(ctrl)
+			vmInterface := kubecli.NewMockVirtualMachineInterface(ctrl)
+			kubecli.MockKubevirtClientInstance.EXPECT().VirtualMachine(testNamespace).Return(vmInterface).AnyTimes()
+			vmInterface.EXPECT().Get(context.Background(), vm.Name, metav1.GetOptions{}).Return(vm, nil).AnyTimes()
```

Comment:
Can you use the kubevirtfake client here?

```suggestion
			virtClient := kubevirtfake.NewSimpleClientset()
			kubecli.MockKubevirtClientInstance.EXPECT().VirtualMachine(testNamespace).
			Return(virtClient.KubevirtV1().VirtualMachines(testNamespace)).AnyTimes()
			vm, err := virtClient.KubevirtV1().VirtualMachines(testNamespace).Create(context.Background(), vm, metav1.CreateOptions{})
			Expect(err).ToNot(HaveOccured())
```